src/Adapter/CombinationDataProvider.php

```diff
@@ -98,8 +109,8 @@ public function getFormCombinations(array $combinationIds, $languageId)
     }
 
     /**
-     * @param $attributesCombinations
-     * @param $product
+     * @param array $attributesCombinations
```

Comment:
Should be `public function completeCombination(array $attributesCombinations, Product$product)` otherwise Doc block is wrong :)

Reply:
it's risky, it's not because this function expect a product, that we always send a Product. I don't know and I won't introduce any regression in this pull request ;)